lib/ecto_adapters_dynamodb/migration.ex

```diff
@@ -174,15 +181,15 @@ defmodule Ecto.Adapters.DynamoDB.Migration do
     to_create = case table.options[:global_indexes] do
       nil -> nil
       global_indexes ->
-        Enum.filter(global_indexes, fn index -> index[:keys] |> Enum.all?(fn key -> Keyword.has_key?(key_list, key) end) end)
+        Enum.filter(global_indexes, fn index -> if index[:keys], do: index[:keys] |> Enum.all?(fn key -> Keyword.has_key?(key_list, key) end) end)
```

Comment:
Discussed this line with Darren just now, and it seems like this change is not necessary. Commenting just so we don't miss reverting and testing.

Reply:
Actually, it turns out that this is necessary - only create actions will have a value for `index[:keys]`, so when we filter while assessing indexes to create, we want to check to make sure that the `:key` key is present.